graph/connected_components.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+class graph
+{
+  private:
+    vector<vector<int>> adj;
+    int cc;
+    void dfs();
+    void explore(int, vector<bool>&);
+  public:
+    graph(int n): adj(n,vector<int>()), cc(0) {}
+    void addEdge(int,int);
+    int getConnectedComponents()
+    {
+      dfs();
+      return cc;
+    }
+};
+
+void graph::addEdge(int u, int v)
+{
+  adj[u-1].push_back(v-1);
+  adj[v-1].push_back(u-1);
+}
+
+void graph::dfs()
```

Comment:
Please rename this function to `depth_first_search()` so the reader does not need to guess.

Reply:
okay